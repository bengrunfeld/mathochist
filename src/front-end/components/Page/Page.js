import React from 'react'
import PropTypes from 'prop-types'
import { Route } from 'react-router-dom'
import Post from '../Post'


const Page = ({postId}) => (
  <section className='col-11 col-lg-9 offset-lg-1 pt-5 mb-5'>
    <Route exact path={`/posts/${postId}`} render={() => <Post postId={postId} />} />
    {/* <Route path='/about' component={About} /> */}
    {/* <Route path='/project' component={Project} /> */}
  </section>
)

Page.propTypes = { postId: PropTypes.string }

export default Page

// How to pass props
// <Route path="/posts" render={() => <Post title='hi' />}/>