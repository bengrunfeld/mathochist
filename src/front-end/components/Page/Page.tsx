import React from 'react'
import { Route, Switch } from 'react-router-dom'
import Post from '../Post'
import About from '../About'
import HomePage from '../HomePage'
import Projects from '../Projects'
import NotFound404 from '../NotFound404'


const Page = () => (
  <section className='col-11 col-lg-9 offset-lg-1 pt-5 mb-5'>
    <Switch>
      <Route exact path='/' component={HomePage} />
      <Route exact path='/page/about' component={About} />
      <Route path='/page/projects' component={Projects} />
      <Route exact path='/post/react' render={() => <Post postId='react' />} />
      <Route exact path='/post/redux' render={() => <Post postId='redux' />} />
      <Route exact path='/post/immutable' render={() => <Post postId='immutable' />} />
      <Route component={NotFound404} />
    </Switch>
  </section>
)


export default Page

// How to pass props
// <Route path="/posts" render={() => <Post title='hi' />}/>