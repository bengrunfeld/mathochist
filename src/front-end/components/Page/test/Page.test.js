import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Page from '../Page'
import setUpTree from '../../../../test/setUpTree'


describe('<Page />', () => {
  test('matches the snapshot', () => {
    const tree = setUpTree(Page, MemoryRouter, mount, '/')
    expect(tree).toMatchSnapshot()
  })
})
