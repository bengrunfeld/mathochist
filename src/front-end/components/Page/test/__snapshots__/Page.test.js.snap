// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> matches the snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Page>
      <section
        className="col-11 col-lg-9 offset-lg-1 pt-5 mb-5"
      >
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              }
            }
            path="/"
          >
            <HomePage
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div>
                <h1>
                  Home Page
                </h1>
              </div>
            </HomePage>
          </Route>
        </Switch>
      </section>
    </Page>
  </Router>
</MemoryRouter>
`;
